tutorial.md

```diff
@@ -0,0 +1,202 @@
+# Tutorial for scylla-cdc-rust
+
+This document will guide you through all steps of creating an example 
+application using the library. 
+We're going to create a simpler version of the [printer](scylla-cdc-printer). 
+
+For the purpose of this tutorial, we're going to assume that the table
+we want to observe the logs for was created by using the following query:
+
+```cassandraql
+CREATE TABLE ks.t (pk int, ck int, v int, vs set<int>, PRIMARY KEY (pk, ck)) WITH cdc = {'enabled': 'true'}
+```
+
+## Creating the CDC consumer
+The most important part of using the library is to define a callback that will be executed 
+after reading a CDC log from the database. 
+Such callback is defined by implementing the `Consumer` trait located in `scylla-cdc::consumer`. 
+As for now, we will define a struct with no member variables for this purpose:
+```rust
+struct TutorialConsumer;
+```
+Since the callback will be executed asynchronously, 
+we have to use the [async-trait](https://crates.io/crates/async-trait) crate 
+to implement the `Consumer` trait. 
+We also use the [anyhow](https://crates.io/crates/anyhow) crate for error handling.
+```rust
+use anyhow;
+use async_trait::async_trait;
+use scylla_cdc::consumer::*;
+
+struct TutorialConsumer;
+
+#[async_trait]
+impl Consumer for TutorialConsumer {
+    async fn consume_cdc(&mut self, _data: CDCRow<'_>) -> anyhow::Result<()> {
+        println!("Hello, scylla-cdc!");
+        Ok(())
+    }
+}
+```
+
+We'll cover on how to use the `CDCRow` structure in [Using CDCRow](tutorial.md#Using CDCRow).
+
+The library is going to create many instances of `TutorialConsumer`, so we also need to define
+a `ConsumerFactory` for them:
+```rust
+struct TutorialConsumerFactory;
+
+#[async_trait]
+impl ConsumerFactory for TutorialConsumerFactory {
+    async fn new_consumer(&self) -> Box<dyn Consumer> {
+        Box::new(TutorialConsumer)
+    }
+}
+```
+
+### Adding shared state to the consumer
+A struct implementing the `Consumer` trait should also implement `Send` trait 
+and a struct implementing the `ConsumerFactory` trait should implement `Send` and `Sync` traits.
+Luckily, Rust implements these traits by default if all member variables of a struct implement them.
```

Comment:
For context, as to _why_ those traits are required, we should mention that the consumer logic is run in separate tokio tasks and the runtime may schedule them on separate threads.